src/utils/self_test/self_test.c

```diff
@@ -434,72 +695,1112 @@ static void print_results(struct st_latency *latencies,
+						key, compar_result);
+increment_code:
+				/* loop control:*/
+				j++;
+				if (j == number_keys) {
+					j = 0;
+					number_keys = ConfigGetKeys(cfg_output,
+							    section_name,
+							    key_names, max,
+							    total_number_keys);
+					total_number_keys += number_keys;
+				}
+			}
+		} /* end of valid test loop */
+	} /* end of "for" loop */
+cleanup:
+	if (cfg_expected != NULL)
+		ConfigFree(cfg_expected);
+	return ret_value;
+}
+
+/* Place output results into output file, tagging with master key */
+static int combine_results(Config *cfg_results, char *section_name)
+{
+	ConfigRet	 ret;
+	int		 ret_value = 0;
+	int		 temp;
+	int		 temp2;
+	int		 i;
+	int		 dfault = -1;
+	const char	*key_name;
+	size_t		 size = MASTER_VALUE_SIZE;
+	char		 new_key_name[2 * MASTER_VALUE_SIZE];
+	char		 master[MASTER_VALUE_SIZE];
+	int		 status_size = sizeof(status) /
+				       sizeof(struct status_feature);
+
+	/* Read master-endpoint string */
+	ret = ConfigReadString(cfg_results, section_name, "master_endpoint",
+			       master, size, "ME");
+	/*
+	 * Copy results into output.
+	 * If the output already has it, then remove it.
+	 */
+	for (i = 0; i < status_size; i++) {
+		key_name = status[i].name;
+		ret = ConfigReadInt(cfg_results, section_name, key_name,
+				    &temp, dfault);
+		/* Tag master endpoint to key name */
+		snprintf(new_key_name, sizeof(new_key_name),
+			 "%s-%s", master, key_name);
+
+		/* Check to see if key occurres in results */
+		if (ret == CONFIG_OK) {
+			/* Check to see if key already exist in output */
+			ret = ConfigReadInt(cfg_output, section_name,
+					    key_name, &temp2, dfault);
+			if (ret == CONFIG_OK) {
+				ConfigRemoveKey(cfg_output, section_name,
+						key_name);
+			}
+			ConfigAddInt(cfg_output, section_name,
+				     new_key_name, temp);
+		}
+	}
+
+#ifdef PRINT_IMMEDIATE_RESULTS
+#define PRINT_IMMEDIATE_RESULTS
+	printf(" Print input results\n");
+	ConfigPrintSection(cfg_results, stdout, section_name);
+
+	printf("\n Output file results\n");
+	ConfigPrintSection(cfg_output, stdout, section_name);
+#endif
+
+	/* print out results if file specified */
+	if (g_expected_outfile != NULL) {
+		/* avoid checkpatch warning */
+		ret = ConfigPrintToFile(cfg_output, g_expected_outfile);
+		if (ret != CONFIG_OK) {
+			D_ERROR("Fail to write to output file: %s\n",
+				g_expected_outfile);
+		ret_value = -ENOENT;
+		}
+	}
+	return ret_value;
+}
+
+static int
+file_name_create(char **gpath_name, bool *name_allocated, char *env)
+{
+	char		*new_name = NULL;
+	char		*env_name;
+	char		*file_name;
+	char		*dup_name;
+	char		*temp_name = NULL;
+	int		 t_length;
+	int		 ret;
+	int		 ret_value = 0;
+	char		*path_name = *gpath_name;
+
+	/* get env to append, if not defined, then leave name as it is */
+	env_name = getenv(env);
+
+	D_EMIT(" Path name: %s\n", path_name);
+	D_EMIT(" env:       %s\n", env_name);
+
+	if (env_name == NULL) {
+		D_EMIT(" Environment %s not set\n", env);
+		goto cleanup;
```

Comment:
there is a mix of goto vs D_GOTO statements throughout the code; it is best if we stick with one

Reply:
All goto are changed to D_GOTO.